ambry-router/src/main/java/com/github/ambry/router/GetBlobOperation.java

```diff
@@ -1751,11 +1799,14 @@ private void handleSimpleBlob(BlobData blobData, byte[] userMetadata, ByteBuffer
             numChunksTotal = 1;
             boolean launchedJob = maybeLaunchCryptoJob(chunkBuf, userMetadata, encryptionKey, blobId);
             if (!launchedJob) {
-              chunkBuf = filterChunkToRange(chunkBuf);
               routerMetrics.getFirstDataChunkLatencyMs.update(System.currentTimeMillis() - submissionTimeMs);
               routerMetrics.getDataChunkLatencyMs.update(System.currentTimeMillis() - initializedTimeMs);
-              chunkIndexToBuf.put(0, chunkBuf.retainedDuplicate());
-              numChunksRetrieved.set(1);
+
+              ByteBuf decompressedContent = decompressContent(chunkBuf.retainedDuplicate());
```

Comment:
Same as above. Do we probably need to decompress content before trying to resolve byte range in line 1789?

Reply:
Yes, we must decompress first before applying range filtering.